pkg/ingester/ingester.go

```diff
@@ -459,6 +459,20 @@ func (i *Ingester) QueryStream(req *client.QueryRequest, stream client.Ingester_
 		return err
 	}
 
+	now := model.Now()
+	if from.After(now) {
+		// time-span start is in future ... regard as legal
+		level.Info(log).Log("msg", "whole timerange in future, yield empty resultset", "through", through, "from", from, "now", now)
+		return nil
+	}
+
+	maxFutureThrough := now.Add(5 * time.Minute)
```

Comment:
I think would be safe if replace `5 * time.Minute` with a shared constant between ingester and chunk store, to keep consistency.